python/lsst/meas/extensions/scarlet/scarletDeblendTask.py

```diff
@@ -527,8 +429,8 @@ class ScarletDeblendConfig(pexConfig.Config):
         dtype=str,
         default="lite",
         allowed={
-            "scarlet": "main scarlet version (likely to be deprecated soon)",
```

Comment:
Agreed. In this case `version=scarlet` was only for testing and was never used anywhere in production. I would also argue that the `oldScarletToData` can be safely removed once this PR is accepted since it only exists to validate this ticket.